### PR TITLE
Add publish button example

### PR DESCRIPTION
**Pull request by azer**

### CODE REVIEW

File: samples/publish-confirm/extension.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "id": "custom-publish",
+  "name": "Publish",
+  "srcdoc": "./index.html",
+  "sidebar": true,
+  "fieldTypes": ["Symbol"]
```
Comment (suevalov) [inline samples/publish-confirm/extension.json]: It became optional after latests @jelz changes.

File: samples/publish-confirm/index.jsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  Subheading
+} from "@contentful/forma-36-react-components"
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.onClickPublish = this.onClickPublish.bind(this)
+    this.onClickUnpublish = this.onClickUnpublish.bind(this)
+    this.onUpdate = this.onUpdate.bind(this)
+    this.onError = this.onError.bind(this)
```
Comment (suevalov) [inline samples/publish-confirm/index.jsx]: I would use class properties which is more common approach in community. Can be enabled by using additional Babel plugin

File: samples/publish-confirm/index.jsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+    // Update component state when a field value changes
+    const fields = this.props.extension.entry.fields
+    for (let key in fields) {
+      fields[key].onValueChanged(this.onUpdate)
+    }
```
Comment (suevalov) [inline samples/publish-confirm/index.jsx]: It looks like this fields callbacks binding should happen only on `componentDidMount`.

File: samples/publish-confirm/index.jsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+    // Trigger update method to setup initial state
+    this.state = this.constructState()
+  }
+
+  constructState() {
+    const sys = this.getEntrySys()
```
Comment (suevalov) [inline samples/publish-confirm/index.jsx]: `this.getEntrySys()` is used only this this function, so I would inline the logic to make it easier to follow.

File: samples/publish-confirm/index.jsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    return {
+      isDraft: !sys.publishedVersion,
+      hasPendingChanges: sys.version > (sys.publishedVersion || 0) + 1,
+      isPublished: sys.version === (sys.publishedVersion || 0) + 1
+    }
+  }
+
+  onError(error) {
+    this.props.extension.dialogs.openAlert({
+      title: "An error happened",
```
Comment (suevalov) [inline samples/publish-confirm/index.jsx]: We have notifications API in SDK which allows us to show Forma36 notifications.

File: samples/publish-confirm/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@contentful/forma-36-fcss": "0.0.9",
+    "@contentful/forma-36-react-components": "^2.3.0",
+    "babel-polyfill": "^6.26.0",
+    "contentful-ui-extensions-sdk": "^3.5.0",
+    "react": "^16.7.0",
+    "react-dom": "^16.7.0",
+    "relative-date": "github:azer/relative-date"
```
Comment (suevalov) [inline samples/publish-confirm/package.json]: please, don't use your own modules, especially that are hosted on github

File: samples/publish-confirm/package.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  "name": "publish-confirm",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "parcel src/index.html"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
```
Comment (jelz) [inline samples/publish-confirm/package.json]: remove all deps we reference from unpkg

File: samples/publish-confirm/package.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    "@contentful/forma-36-fcss": "0.0.9",
+    "@contentful/forma-36-react-components": "^2.3.0",
+    "babel-polyfill": "^6.26.0",
```
Comment (jelz) [inline samples/publish-confirm/package.json]: shouldn't be https://www.npmjs.com/package/@babel/polyfill ?

File: samples/publish-confirm/package.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    "contentful-ui-extensions-sdk": "^3.5.0",
+    "react": "^16.7.0",
+    "react-dom": "^16.7.0",
+    "relative-date": "^1.1.3"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.2.2",
+    "@babel/plugin-proposal-class-properties": "^7.3.0",
+    "cssnano": "^4.1.8",
```
Comment (jelz) [inline samples/publish-confirm/package.json]: this is not used. I also think parcel doesn't require `@babel` dependencies you've declared.

File: samples/publish-confirm/package.json [hunk 10]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "publish-confirm",
+  "version": "1.0.0",
+  "description": "",
```
Comment (jelz) [inline samples/publish-confirm/package.json]: provide a description

File: samples/publish-confirm/package.json [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  "main": "index.js",
+  "scripts": {
+    "start": "parcel src/index.html"
+  },
+  "author": "",
```
Comment (jelz) [inline samples/publish-confirm/package.json]: author should be "Contentful GmbH", license "MIT"

**Pull request merged**